src/request_handler.rs

```diff
@@ -1,21 +1,21 @@
 use std::str;
 use std::sync::Arc;
 
-use hyper::rt::{Future, Stream};
+use futures_util::stream::StreamExt;
 use hyper::{Body, Method, Request, Response, StatusCode, Uri};
 use parking_lot::RwLock;
 
 use crate::component::ComponentManager;
 use crate::error::{WorkerError, WorkerErrorKind};
-use crate::model::ComponentPath;
+use crate::model::{ComponentPath, StatusColor};
 
 // Warning: This method is somewhat complicated, since it needs to deal with async stuff
 // There should be no state here beyond the handler, so no need for an actual hyper service
 // (We don't want to lock into hyper that hard anyway)
-pub fn global_request_entrypoint(
+pub async fn global_request_entrypoint(
```

Comment:
This function seems a lot cleaner with the change to it being async. Do you predict anything has actually changed performance-wise? Or is it similar

Reply:
I think it'll be about the same. I'm kinda annoyed about how getting the entire body as a future is so hard -- kinda wish there was a better solution upstream.

Reply:
A lot of deeper stuff can become `async` in a future refactor, which definitely improves parallelism / is exciting!